packages/web-frontend/cypress/scripts/testE2e/testE2e.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { execSync } from 'child_process';
+import {} from 'dotenv/config';
+import GithubApi from 'github-api';
+
+import { getLoggerInstance, RemoteGitRepo, requireEnv } from '@tupaia/utils';
+import { SNAPSHOTS } from '../../constants';
+import { pullSnapshots } from './pullSnapshots';
+import { pushSnapshots } from './pushSnapshots';
+import { Snapshots } from './Snapshots';
+
+const runPackageScript = script =>
+  execSync(`yarn workspace @tupaia/web-frontend ${script}`, { stdio: 'inherit' });
+
+const runTestsAndCatchErrors = () => {
+  const record = !!process.env.CYPRESS_RECORD_KEY;
+
+  let cypressError;
+  try {
+    runPackageScript(`cypress:run --record ${record}`);
+  } catch (error) {
+    // Note: ideally we would only catch test assertion errors,
+    // and still throw runtime/syntax errors etc.
+    // Unfortunately we haven't found a eay to distinguish between the two in this occasion
+    cypressError = error;
+  }
+
+  return cypressError;
+};
+
+const getCurrentLocalBranch = () =>
+  execSync('git rev-parse --abbrev-ref HEAD', { encoding: 'utf-8', stdio: 'pipe' }).split('\n')[0];
+
+export const getSnapshotBranch = () => {
+  const branch = process.env.CI_BRANCH || getCurrentLocalBranch();
+  return branch.replace(/-e2e$/, '');
+};
+
+export const getSnapshotRepo = () => {
+  const gitHub = new GithubApi({ token: requireEnv('GITHUB_ACCESS_TOKEN') });
+  return RemoteGitRepo.connect(gitHub, SNAPSHOTS.repoUrl);
+};
+
+/**
+ * Runs e2e tests using the following steps:
+ * 1. Pull snapshots
+ * 2. Generate config
+ * 3. Run tests using the pulled snapshots
+ * 4. If new snapshots were captured during tests, push them
```

Comment:
```suggestion
 * 4. If new snapshots were captured during tests, or differences in output found, push them
```

Reply:
From what I can see, this is correct? Generally we hope they will just be new snapshots, but sometimes they will be differences that are either legit or not legit, in which case they will be pushed for manual review right?

Reply:
Most definitely! The truth is, "new" is probably a misnomer in this PR, a more accurate definition would be "snapshotsCapturedDuringLastTestRun". But couldn't find a short and elegant description for that 😆 

I should definitely update this comment though. Please let me know if you come up with a better term than "new" snapshtos!